conductor_api/src/conductor/test_admin.rs

```diff
@@ -0,0 +1,57 @@
+use crate::{
+    conductor::{base::notify, Conductor},
+    config::AgentConfiguration,
+    key_loaders::test_keystore,
+    keystore::PRIMARY_KEYBUNDLE_ID,
+};
+use holochain_core_types::error::HolochainError;
+
+pub trait ConductorTestAdmin {
+    fn add_test_agent(&mut self, id: String, name: String) -> Result<String, HolochainError>;
+}
+
+impl ConductorTestAdmin for Conductor {
+    fn add_test_agent(&mut self, id: String, name: String) -> Result<String, HolochainError> {
+        let mut new_config = self.config.clone();
+        if new_config.agents.iter().any(|i| i.id == id) {
+            return Err(HolochainError::ErrorGeneric(format!(
+                "Agent with ID '{}' already exists",
+                id
+            )));
+        }
+        let mut keystore = test_keystore(&name);
+        let keybundle = keystore
+            .get_keybundle(PRIMARY_KEYBUNDLE_ID)
+            .expect("Couldn't get KeyBundle that was just added back from Keystore");
+        let public_address = keybundle.get_id();
+        let new_agent = AgentConfiguration {
+            id: id.clone(),
+            name: name.clone(),
+            public_address: public_address.clone(),
+            keystore_file: name.clone(),
+            holo_remote_key: None,
+        };
+        new_config.agents.push(new_agent);
+        new_config.check_consistency()?;
+        self.config = new_config;
+        // self.save_config()?; we don't actually want to save it for tests
```

Comment:
we could just pull this line out?

Reply:
Well it might be good to keep in because it reminds us that we Don't actually want to save the config, unlike in other code chunks..